tests/scripts/thread-cert/node.py

```diff
@@ -175,6 +175,55 @@ def _expect(self, pattern, timeout=-1, *args, **kwargs):
                 if timeout <= 0:
                     raise
 
+    def _prepare_pattern(self, pattern):
+        """Build a new pexpect pattern matching line by line.
+
+        Adds lookahead and lookbehind to make each pattern match a whole line,
+        and add 'Done' as the first pattern.
+
+        Args:
+            pattern: a single regex or a list of regex.
+
+        Returns:
+            A list of regex.
+        """
+        EXPECT_LINE_FORMAT = r'(?<=[\r\n])%s(?=[\r\n])'
```

Comment:
This regex is not totally correct. If `pattern` == `''` (empty string) , then the line prepared pattern will be `r'(?<=[\r\n])(?=[\r\n])'`. We are expecting it to match an empty line between 2 EOLs, e.x. `'\n\n'`, `'\r\n\r\n'`. However, it will also match `'\r\n'`, which is not an empty line between two EOLs.

Reply:
Do we have a use case to expect a empty line? There is a limitation that look behind has to be with fixed size. And with readline enabled, there are mixed \r\n, \n, even \r as new line delimiter.

Reply:
Do you think `r'(?<=[\n])%s(?=\r?\n)' ` will be better? Since `node` is reading from CLI executive what is written by us, we can be sure that newline always ends with `\n`.

Reply:
IIRC when readline is enabled, sometimes a line starts with a `\r`.

Reply:
SGTM since we have no use case to expect an empty line yet.